### ISSUE TITLE
Improve download time for large lists of files

### ISSUE THREAD

**Issue by andresdavid90** [andresdavid90/pizza-guy#56]
Improve the `download-pool` internal module to make use of any available slot after it is immediately available at the end of each download.

Right now, we are downloading 6 files at the same time and waiting the 6 files to finish before request the next 6. Ideally, we should use an available slot right after it becomes free.

This should give us at least in theory, faster downloads.

@baldore @alejandronanez does this makes sense for you guys?

**Comment by baldore** [comment]
Yeah! sounds good. Also, it should be great to set the limit in the configuration and use 6 by default.

In that case, we need to modify the logic that we have right now, because this behavior is totally different (Splice won't be required anymore, since we don't need to separate the list of files into bundles, but instead limit the downloads). 

Finally, I don't know how are we gonna test this...

**Comment by andresdavid90** [comment]
We need to try it with some POCs to see how this can behave.
